client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -149,6 +149,26 @@ export class ShippingBanner extends Component {
 		}
 	}
 
+	generateMetaBoxHtml( title, args ) {
+		const argsJsonString = JSON.stringify( args ).replace( /"/g, '&quot;' ); // JS has no native html_entities so we just replace.
+
+		const togglePanelText = __( 'Toggle panel:', 'woocommerce-admin' );
+
+		return `
+<div id="woocommerce-order-label" class="postbox ">
+	<button type="button" class="handlediv" aria-expanded="true">
+		<span class="screen-reader-text">${ togglePanelText } ${ title }</span>
+		<span class="toggle-indicator" aria-hidden="true"></span>
+	</button>
+	<h2 class="hndle"><span>${ title }</span></h2>
+	<div class="inside">
+		<div class="wcc-root woocommerce wc-connect-create-shipping-label" data-args="${ argsJsonString }">
```

Comment:
Since `generateMetaBoxHtml()` is rendering HTML. I was thinking maybe we could change  `generateMetaBoxHtml()` into a React component which it will render the same HTML by passing the parameter as props in.

The tricky part is how we could do `.insertAdjacentHTML( 'beforebegin', shippingLabelContainerHtml );` in React. 

I think having composited components in React-land is better than inserting raw HTML directly into the DOM. But even if we have the component, we still need to do `.insertAdjacentHTML()` kinda thing to insert these meta box directly in the DOM. I am not sure what benefits we will get by converting these HTML to a component.

I am 👍  with this but you may want others' opinions.

Reply:
I think the best option is actually to load the html via an ajax endpoint. That way any changes in the future in the WCS codebase would be reflected here. That would mean we would need another release of WCS and that seems like too much.